src/Libraries/Watch3DNodeModels/Watch3D.cs

```diff
@@ -95,9 +95,9 @@ public Watch3DCamera()
     [OutPortTypes("var")]
     [AlsoKnownAs("Dynamo.Nodes.dyn3DPreview", "Dynamo.Nodes.3DPreview", "Dynamo.Nodes.Watch3D", "DynamoWatch3D.Watch3D")]
     [IsDesignScriptCompatible]
-    public class Watch3D : NodeModel
+    public class Watch3D : VariableInputNode
```

Comment:
This is an API break. This is a public type, and introducing a new intermediate base class which is abstract forces the implementor to implement two new methods.

Previous inheritors of this class will now likely have broken code at runtime.

Now it's unlikely that someone has inherited from Watch3d - but we have no idea and so this would force us to update to Dynamo 3.x. Anytime we decide to break API compatibility/semantic versioning for some reason a decision/discussion is required.

It's difficult to judge if the change is trivial or who it will affect, we can really use our best judgement.

To avoid those discussions a pattern can be used, like was done in the new `if` node work.
https://github.com/DynamoDS/Dynamo/pull/11728/files

the old node is obsoleted and an `[AlsoKnownAs]` attribute is added to the new node so migration is performed. You can also create more complex migrations using the migration framework to replace multiple nodes etc - like the List.Create.

Reply:
I am considering this solution: implementing a intermediate non-abstract class between Watch3D and VariableInputNode. This class will take care of the two required abstract methods from VariableInputNode and any other aspects so that the Watch3D node will still behave as it currently does (have not thought this through fully yet). Please Let me know your thoughts.

Reply:
@mjkkirschner I just added a new BasicVariableInputNode class with default implementations for the two abstract methods. Will this be an acceptable solution?

Reply:
I don't know - did you look into the .net api compatibility documentation ?

https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules